python/tank/bootstrap/configuration_writer.py

```diff
@@ -213,26 +214,60 @@ def move_to_backup(self, undo_on_error):
             return (config_backup_path, core_backup_path)
 
     @filesystem.with_cleared_umask
-    def create_tank_command(self, win_python=None, mac_python=None, linux_python=None):
+    def create_tank_command(self):
         """
         Create a tank command for this configuration.
 
         The tank command binaries will be copied from the current core distribution
-        The interpreter_xxx.cfg files will be created based on the given interpreter settings.
-        Uses sg desktop python if no interpreter paths are provided.
+        The interpreter_xxx.cfg files will be created based on the ``sys.executable``.
 
-        :param win_python: Optional path to a python interpreter
-        :param mac_python: Optional path to a python interpreter
-        :param linux_python: Optional path to a python interpreter
+        :param current_interpreter: Path to the current interpreter. Defaults to sys.executable.
         """
         log.debug("Installing tank command...")
 
         # first set up the interpreter_xxx files needed for the tank command
         # default to the shotgun desktop python
-        executables = {}
-        executables["Linux"] = linux_python or constants.DESKTOP_PYTHON_LINUX
-        executables["Darwin"] = mac_python or constants.DESKTOP_PYTHON_MAC
-        executables["Windows"] = win_python or constants.DESKTOP_PYTHON_WIN
+        executables = dict(
+            Linux="",
+            Darwin="",
+            Windows=""
+        )
+
+        # FIXME: This is a really bad hack. We're looking to see if we are running inside the
+        # Shogun Desktop and if we are then we're using the Python that is package with it.
+        #
+        # The reason we are doing this is because we need the interpreter files
+        # written out during bootstrapping to be using the ones from the Shotgun Desktop.
+        #
+        # We could have introduced a way on the ToolkitManager to specify the interpreter
+        # to use for the current platform for descriptor based configuration, but we're trying
+        # to kill the interpreter files in the first place mid-term, so introducing an API
+        # that allows the caller to specify which one to use feels backwards in the first place.
+        #
+        # This feels like the lesser of two evils, even tough as I type this I've thrown
+        # up a bit in my mouth.
+
+        # Figures out which is the current Python interpreter.
+        if os.path.split(sys.executable)[1].lower().startswith("shotgun"):
+            log.debug("Shotgun Desktop process detected.")
+            if sys.platform == "darwin":
+                current_interpreter = os.path.join(sys.prefix, "bin", "python")
+            elif sys.platform == "win32":
+                current_interpreter = os.path.join(sys.prefix, "python.exe")
+            else:
+                current_interpreter = os.path.join(sys.prefix, "bin", "python")
+        else:
+            current_interpreter = sys.executable
```

Comment:
use case: you have a maya config in `~/Library/caches/Shotgun/....` and want to troubleshoot via the `tank` command. if we use sys.executable, interpreter file will probably have the path to maya. In that case, better to have the path to the default install of desktop python. Won't work for clients like milk, but works for most studios. We then discuss a strategy to improve this further.